src/classes/UTIL_Namespace.cls

```diff
@@ -98,17 +98,20 @@ public class UTIL_Namespace {
     
     /*******************************************************************************************************
     * @description Static method that takes a string of a multiple potential field names or object names.  
-    * If it is a custom object or field (name ends with __c), it prepends the namespace prefix if required.
+    * If it is a custom object or field (name ends with __c, __r or __mdt), it prepends the namespace prefix if required.
     * If we are in a managed package, tokens in dynamic SOQL must include the package namespace prefix.
     * If you ever deploy this package as unmanaged, this routine will do nothing!
     * @param str string that contains 0 or more token names 
     * @return same string with token names, namespace prefixed, if required.
     ********************************************************************************************************/
     public static string StrAllNSPrefix(string str) {
         if (getNamespace() == '') return str;
-        Pattern pat = Pattern.Compile('[a-zA-z0-9]*__c');
-        Matcher match = pat.matcher(str);
-        return match.replaceAll(getNamespace()+'__$0');
+        Pattern pat1 = Pattern.Compile('[a-zA-z0-9]*__[cr]');
```

Comment:
I believe these can be combined into a single  ```Pattern.Compile('[a-zA-z0-9]*__(?:c|r|mdt)');```

Note: I haven't tested the above regex :)

Reply:
Worked perfectly! Regex is definitely my weakness!